ideascube/serveradmin/catalog.py

```diff
@@ -762,21 +763,26 @@ def update_cache(self):
 
         for remote in self._remotes.values():
             # TODO: Get resumable.urlretrieve to accept a file-like object?
-            fd, tmppath = tempfile.mkstemp()
-            os.close(fd)
+            with tempfile.NamedTemporaryFile() as fd:
+                tmppath = fd.name
 
-            def _progress(*args):
-                self._progress(' {}'.format(remote.name), *args)
+                def _progress(*args):
+                    self._progress(' {}'.format(remote.name), *args)
```

Comment:
Maybe keep the definition outside of the `with` block?

Less lines changed, less stuff to review. :stuck_out_tongue: